src/dashboard/src/components/administration/views_processing.py

```diff
@@ -15,404 +15,64 @@
 
-    select_fields = chain_choice_fields + replace_dict_fields + storage_service_options
 
+def edit(request, name=None):
     if request.method == 'POST':
-        # render XML using request data
-        xmlChoices = PreconfiguredChoices()
-
-        # use toggle field submissions to add to XML
-        for field in boolean_select_fields:
-            enabled = request.POST.get(field['name'])
-            if enabled == 'yes':
-                if 'yes_option' in field:
-                    # can be set to either yes or no
-                    toggle = request.POST.get(field['name'] + '_toggle', '')
-                    if toggle == 'yes':
-                        go_to_chain_text = field['yes_option']
-                    elif 'no_option' in field:
-                        go_to_chain_text = field['no_option']
-
-                    if 'no_option' in field:
-                        xmlChoices.add_choice(
-                            field['choice_uuid'],
-                            go_to_chain_text
-                        )
-                    else:
-                        if toggle == 'yes':
-                            xmlChoices.add_choice(
-                                field['choice_uuid'],
-                                go_to_chain_text
-                            )
-
-        # set quarantine duration if applicable
-        quarantine_expiry_enabled = request.POST.get('quarantine_expiry_enabled', '')
-        quarantine_expiry         = request.POST.get('quarantine_expiry', '')
-        if quarantine_expiry_enabled == 'yes' and quarantine_expiry != '':
-            xmlChoices.add_choice(
-                '19adb668-b19a-4fcb-8938-f49d7485eaf3', # Remove from quarantine
-                '333643b7-122a-4019-8bef-996443f3ecc5', # Unquarantine
-                str(float(quarantine_expiry) * (24 * 60 * 60))
-            )
-
-        # use select field submissions to add to XML
-        for field in select_fields:
-            enabled = request.POST.get(field['name'] + '_enabled')
-            if enabled == 'yes':
-                field_value = request.POST.get(field['name'], '')
-                if field_value != '':
-                    if field.get('do_not_lookup', False):
-                        target = field_value
-                    else:
-                        target = uuid_from_description(field_value, field['choice_uuid'])
-
-                    xmlChoices.add_choice(
-                        field['choice_uuid'],
-                        target
-                    )
-
-        xmlChoices.write_to_file(file_path)
-
-        messages.info(request, 'Saved!')
-
-        return redirect('components.administration.views.processing')
+        form = ProcessingConfigurationForm(request.POST)
+        if form.is_valid():
+            try:
+                form.save_config()
+            except Exception as e:
+                logger.exception(str(e))
```

Comment:
`logger.exception` already includes a traceback for the exception. Maybe make the error message more useful, like 'Failed to save processing config' ie why this traceback happened.